static/app/utils/replays/playback/providers/useReplayPlayerState.tsx

```diff
@@ -0,0 +1,295 @@
+import type {Dispatch} from 'react';
+import {createContext, useCallback, useContext, useReducer} from 'react';
+import type {Replayer} from '@sentry-internal/rrweb';
+import {ReplayerEvents} from '@sentry-internal/rrweb';
+import type {
+  PlayerState,
+  SpeedState,
+} from '@sentry-internal/rrweb/typings/replay/machine';
+
+import type {ReplayPrefs} from 'sentry/components/replays/preferences/replayPreferences';
+import {uniq} from 'sentry/utils/array/uniq';
+import clamp from 'sentry/utils/number/clamp';
+import type {Dimensions} from 'sentry/utils/replays/types';
+
+type ReplayerAction =
+  | {dispatch: Dispatch<ReplayerAction>; replayer: Replayer; type: 'didMountPlayer'}
+  | {replayer: Replayer; type: 'didUnmountPlayer'}
+  | {type: 'didStart'}
+  | {type: 'didPause'}
+  | {type: 'didResume'}
+  | {type: 'didFinish'}
+  | {height: number; type: 'didResize'; width: number}
+  | {speed: number; type: 'didSkipStart'}
+  | {speed: number; type: 'didSkipEnd'}
+  | {type: 'didFlush'}
+  | {playerState: PlayerState; type: 'didPlayerStateChange'}
+  | {speedState: SpeedState; type: 'didSpeedStateChange'};
+
+type UserAction =
+  | {type: 'play'}
+  | {type: 'pause'}
+  | {
+      isSkippingInactive: ReplayPrefs['isSkippingInactive'];
+      type: 'setConfigIsSkippingInactive';
+    }
+  | {playbackSpeed: ReplayPrefs['playbackSpeed']; type: 'setConfigPlaybackSpeed'}
+  | {offsetMs: number; type: 'jumpToOffset'};
+
+interface State {
+  currentSpeed: undefined | number;
+  dimensions: Dimensions;
+  isFinished: boolean;
+  playerState: 'playing' | 'paused' | 'live';
```

Comment:
what's the `live` state?

Reply:
it's when events are streaming in instead of being all pre-loaded up front.

we don't use this mode, i just ported over all the types from the rrweb package for completeness.